### PR TITLE
Assisted ember upgrades

### PR DESCRIPTION
**Pull request by GavinJoyce**
[Rendered](https://github.com/GavinJoyce/rfcs/blob/gj/assisted-upgrades/text/0000-assisted-upgrades.md)

### CODE REVIEW

File: text/0000-assisted-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+process for introducing new syntax and APIs which depreciate old ones should include
+consideration for automating the transformation.
+
+# Drawbacks
+
+Introducing a deprecation would take a little more effort.
+
+# Alternatives
+
+The [ember-watson](https://github.com/abuiles/ember-watson) addon provides some commands for
```
Comment (stefanpenner) [inline text/0000-assisted-upgrades.md]: using ember-watson, and giving it access to the HTMLBars AST transforms etc seems like a reason path forward.

**Comment by ef4** [comment]
I think ember-watson is the right place to solve the problems you're describing.

Last I knew, the barrier to doing htmlbars rewriting was creating a whitespace-preserving AST printer. We actually already implement some of the transformations you'd want, they just happen at compile time now, and can't be easily printed back out as template files.

**Comment by GavinJoyce** [comment]
Fair enough, I'll close this an contribute to [ember-watson](https://github.com/abuiles/ember-watson). Perhaps we can revisit this once it has more comprehensive coverage of depreciations? It would be nice to explicitly include new depreciations in ember-watson as part of the depreciation process, include links in the depreciation documentation and possible include ember-watson in ember-cli

**Comment by rwjblue** [comment]
I don't think we will block new deprecations from being added on having automated upgrade tooling, but we definitely are blocking deprecations without an entry in the deprecation guide for the version in question. My goal would be to ensure that this guide has all the info required for someone to be able to add upgrade automation to ember-watson.

**Comment by GavinJoyce** [comment]
@rwjblue Sounds good, thanks

**Comment by GavinJoyce** [comment]
The first of many (hopefully): https://github.com/abuiles/ember-watson/pull/87

**Pull request closed without merging**